rust/src/http2/range.rs

```diff
@@ -38,52 +40,45 @@ pub struct HTTPContentRange {
 }
 
 pub fn http2_parse_content_range_star<'a>(input: &'a [u8]) -> IResult<&'a [u8], HTTPContentRange> {
-    let (i2, _) = char!(input, '*')?;
-    let (i2, _) = char!(i2, '/')?;
-    let (i2, size) = map_res!(i2, map_res!(digit1, std::str::from_utf8), i64::from_str)?;
+    let (i2, _) = char('*')(input)?;
```

Comment:
Looks strange to me to use `char` as a function.
Is there no risk of future collision ?

Reply:
You mean because `char` is a rust keyword? I don't think there will be problems, it is part of the language. You can even use reserved keywords using the raw identifiers notation (see https://doc.rust-lang.org/rust-by-example/compatibility/raw_identifiers.html).

Note that the previous code was also using keywords as variable names in other places (see `u64` in `detect_parse_u64`)

Reply:
ok fair enough

> see u64 in detect_parse_u64

I feel the shame